packages/flutter/lib/src/widgets/localized_resources.dart

```diff
@@ -0,0 +1,354 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' show Locale;
+
+import 'package:flutter/foundation.dart';
+
+import 'container.dart';
+import 'framework.dart';
+
+/// Progress for a [LocalizedResourcesDelegate.load] method.
+enum LocalizedResourcesStatus {
+  /// Loading has not been started, [LocalizedResourcesDelegate.resourcesFor] will return null.
+  none,
+
+  /// Loading is underway, [LocalizedResourcesDelegate.resourcesFor] will return null.
+  loading,
+
+  /// Loading is complete, [LocalizedResourcesDelegate.resourcesFor] will return a valid non-null value.
+  loaded,
+}
+
+/// An encapsulation of all of the resources to be loaded by a
+/// [LocalizedResources] widget.
+///
+/// Typical applications have one [LocalizedResources] widget which is
+/// created by the [WidgetsApp] and configured with the app's
+/// `localizedResourcesDelegate` parameter.
+///
+/// See also:
+///
+///  * [DefaultLocalizedresorucesDelegate], a simple concrete version of
+///    this class.
+abstract class LocalizedResourcesDelegate {
+  /// Start loading the resources for `locale`. The returned future completes
+  /// when the resources have been loaded and cached.
+  ///
+  /// It's assumed that the this method will create one or more objects
+  /// each of which contains a collection or related resources (typically
+  /// defined by one method per resources). The objects will be retrieved
+  /// by [Type] with [resourcesFor].
+  Future<Null> load(Locale locale);
```

Comment:
This method implies that an instance of this class is uninitialized or partially initialized for some time. This complicates the lifecycle that the user has to deal with, and also leaves the `load` method visible at the point in time where the object is already initialized (and therefore not necessary any more).

Alternative: split this class into two classes - a loader class and a resources class.

Example:

```dart
class LocalizedResourcesLoader {
  Future<LocalizedResourcesDelegate> load(Locale locale);
}

class LocalizedResourcesDelegate {
  T resourcesFor<T>(...);
  bool updateShouldNotify(...);
}
```

`LocalizedResourcesLoader` does not give you `resourcesFor` and so you cannot accidentally access partially initialize resources. OTOH, `LocalizedResourcesDelegate` doesn't have `load` so you don't accidentally (or "just in case") call `load()`. An additional benefit is that you need less documentation (and users brain cycles) for understanding the lifecycle of resource loading. The API automatically guides you through the process. You also wouldn't need the `LocalizedResourcesStatus` enum (other than perhaps, for logging).